.github/workflows/ci.yaml

```diff
@@ -60,28 +60,44 @@ jobs:
         pip install -r tests/requirements-test.txt
         python -m pytest -v -p no:warnings
         deactivate
+  deployment-tests:
```

Comment:
Using [Job ordering](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idneeds) I think we should enforce the job dependencies like if `code-tests `fail, we should not need to proceed to `deployment-tests.` 

Right now from what I understand `code-tests` and 2 `deployment-tests` jobs all start in parallel (if there are enough runners).  In the event that `code-tests` fail, we still need to wait till one of the `deployment-tests` to fail before the whole workflow is complete and considered failed. 

With job ordering we can immediately fail if `code-tests` fail. Maybe this isn't needed, but interested to know what you think.

Reply:
Hmm I guess you're right. I was thinking `fail-fast` should cover us, but I guess that only covers the matrix. I'll add a `needs` statement to do the code checks first. Good suggestion!